tests/common.go

```diff
@@ -1869,7 +1892,147 @@ func ChangeNamespaces(contexts []*scheduler.Context,
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*storkapi.VolumeSnapshotRestore); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*apapi.AutopilotRule); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*corev1.ServiceAccount); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*rbacv1.Role); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*rbacv1.RoleBinding); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*rbacv1.ClusterRole); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*rbacv1.ClusterRoleBinding); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*batchv1beta1.CronJob); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*batchv1.Job); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*corev1.LimitRange); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*networkingv1beta1.Ingress); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*monitoringv1.Prometheus); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*monitoringv1.PrometheusRule); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*monitoringv1.ServiceMonitor); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*corev1.Namespace); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*apiextensionsv1beta1.CustomResourceDefinition); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*apiextensionsv1.CustomResourceDefinition); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*policyv1beta1.PodDisruptionBudget); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*netv1.NetworkPolicy); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*corev1.Endpoints); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*storkapi.ResourceTransformation); ok {
+		clone := *specObj
+		return &clone, nil
+	} else if specObj, ok := spec.(*admissionregistrationv1.ValidatingWebhookConfiguration); ok {
+		clone := *specObj
+		webhooks := make([]admissionregistrationv1.ValidatingWebhook, 0)
+		for i := range specObj.Webhooks {
+			webhook := specObj.Webhooks[i]
+			serviceClone := *specObj.Webhooks[i].ClientConfig.Service
+			webhook.ClientConfig.Service = &serviceClone
+			webhooks = append(webhooks, webhook)
+		}
+		clone.Webhooks = webhooks
+		return &clone, nil
+	}
+
+	return nil, fmt.Errorf("unsupported object while setting namespace: %v", reflect.TypeOf(spec))
+}
+
+func UpdateNamespace(in interface{}, namespaceMapping map[string]string) error {
```

Comment:
How about giving the implementation of UpdateNamespace below a try? I think it could make the code more readable, but please keep in the mind that I have not tested it.

```
func UpdateNamespace(in interface{}, namespaceMapping map[string]string) error {
    supportedTypes := map[reflect.Type]bool{
        reflect.TypeOf(&appsapi.Deployment{}): true,
        reflect.TypeOf(&appsapi.StatefulSet{}): true,
        reflect.TypeOf(&appsapi.DaemonSet{}): true,
        .
        .
        .
    }
    inObjType := reflect.TypeOf(in)
    if supportedTypes[inObjType] {
        obj := reflect.ValueOf(in).Elem().Interface()
        namespace := namespaceMapping[obj.GetNamespace()]
        obj.SetNamespace(namespace)
        return nil
    }
    return fmt.Errorf("unsupported object while setting namespace: %v", inObjType)
}
```

Reply:
How about adding a docstring that describes what the function does?

Reply:
docstring done.
`reflect` based solution doesn't seem possible as it is not working.